File: harmony-v0-arkts/entry/src/main/reference/android/android-v0-compose/app/src/main/java/com/example/v0clone/data/model/JobDictionaryModels.kt
package com.xlwl.AiMian.data.model

import com.google.gson.annotations.SerializedName

data class JobDictionaryCategory(
    val id: String,
    val code: String,
    val name: String,
    val description: String? = null,
    val sortOrder: Int = 0,
    @SerializedName("isActive")
    val isActive: Boolean = true,
    @SerializedName("positions")
    val positions: List<JobDictionaryPosition> = emptyList()
)

data class JobDictionaryPosition(
    val id: String,
    val categoryId: String,
    val code: String,
    val name: String,
    val description: String? = null,
    val sortOrder: Int = 0,
    @SerializedName("isActive")
    val isActive: Boolean = true,
    val tags: List<String> = emptyList()
)

File: iOS-v0/Sources/Models/JobDictionaryModels.swift
import Foundation

struct JobDictionaryCategory: Codable, Identifiable, Hashable, Sendable {
  let id: String
  let code: String
  let name: String
  let description: String?
  let sortOrder: Int
  let isActive: Bool
  let positions: [JobDictionaryPosition]
}

struct JobDictionaryPosition: Codable, Identifiable, Hashable, Sendable {
  let id: String
  let categoryId: String
  let code: String
  let name: String
  let description: String?
  let sortOrder: Int
  let isActive: Bool
  let tags: [String]
}
